app/components/inspector/SimulatorCard.tsx

```diff
@@ -81,11 +92,97 @@ function useSimulator(message: VersionedMessage) {
         const connection = new Connection(url, 'confirmed');
         (async () => {
             try {
-                // Simulate without signers to skip signer verification
+                // Fetch all the accounts before simulating
+                const accountKeys = message.getAccountKeys().staticAccountKeys;
+                const parsedAccountsPre = await connection.getMultipleParsedAccounts(accountKeys);
+
+                // Simulate without signers to skip signer verification. Request
+                // all account data after the simulation.
                 const resp = await connection.simulateTransaction(new VersionedTransaction(message), {
                     replaceRecentBlockhash: true,
+                    accounts: {
+                        encoding: 'base64',
+                        addresses: accountKeys.map(function (key) {
+                            return key.toBase58();
+                        }),
+                    },
                 });
 
+                const mintToDecimals: { [mintPk: string]: number} =  getMintDecimals(
+                    accountKeys,
+                    parsedAccountsPre.value,
+                    resp.value.accounts as SimulatedTransactionAccountInfo[],
+                )
+
+                const preTokenBalances: TokenBalance[] = [];
+                const postTokenBalances: TokenBalance[] = [];
+                const tokenAccountKeys: ParsedMessageAccount[] = [];
+
+                for (let index = 0; index < accountKeys.length; index++) {
+                    const key = accountKeys[index];
+                    const parsedAccountPre = parsedAccountsPre.value[index];
+                    const accountDataPost = resp.value.accounts?.at(index)?.data[0];
+                    const accountOwnerPost = resp.value.accounts?.at(index)?.owner;
+
+                    if (
+                        parsedAccountPre?.owner.toBase58() == TOKEN_PROGRAM_ID.toBase58() &&
```

Comment:
I think we should include the token22 program here too. Eg test with a BERN transaction (mint is CKfatsPMUf8SkiURsDXs7eK6GWb4Jsd6UDbs7twMCWxo).

But note these all seem to use address table lookups, so we probably won't be able to test one yet